jobs/integration/utils.py

```diff
@@ -118,55 +118,41 @@ async def upgrade_charms(model, channel, tools):
     await tools.juju_wait()
 
 
-async def upgrade_snaps(model, channel, tools):
-    for app_name, blocking in {
-        "kubernetes-control-plane": True,
-        "kubernetes-worker": True,
-        "kubernetes-e2e": False,
-    }.items():
+async def upgrade_snaps(model: Model, channel, tools):
+    for app_name in [
+        "kubernetes-control-plane",
+        "kubernetes-worker",
+        "kubernetes-e2e",
+    ]:
```

Comment:
we don't need to know whether or not to expect the applications to block when they undergo a snap upgrade... all we really need to check is 
1) Do the units go idle after the config change (not necessarily active/idle)
2)  is the unit blocked specifically because it needs a manual upgrade

if so -- run the upgrade action.